test/game/logic/item_logisticsTests.cpp

```diff
@@ -341,7 +365,42 @@ namespace jactorio::game
 		asm_data.ingredientInv[1] = {nullptr, 0, recipe_pack.item2};
 
 		// 49 + 2 > 50
-		EXPECT_FALSE(InsertAssemblyMachine(logicData_, {recipe_pack.item1, 2}, asm_data, data::Orientation::up));
+		EXPECT_FALSE(InsertAssemblyMachine({logicData_, {recipe_pack.item1, 2}, asm_data, data::Orientation::up}));
+	}
+
+	TEST_F(ItemDropOffTest, CanInsertAssemblyMachine) {
+		data::AssemblyMachineData asm_data{};
+
+		data::Item item{};
+
+		// No recipe
+		EXPECT_FALSE(CanInsertAssemblyMachine({logicData_, {&item, 2}, asm_data, data::Orientation::down}));
+
+
+		data::PrototypeManager prototype_manager{};
+
+		auto recipe_pack = TestSetupRecipe(prototype_manager);
+		asm_data.ChangeRecipe(logicData_, prototype_manager, &recipe_pack.recipe);
+
+		// Has recipe, wrong item
+		EXPECT_FALSE(CanInsertAssemblyMachine({logicData_, {&item, 2}, asm_data, data::Orientation::down}));
+
+		// Has recipe, correct item
+		DropOffParams args{logicData_, {recipe_pack.item1, 2000}, asm_data, data::Orientation::down};
+		EXPECT_TRUE(CanInsertAssemblyMachine(args));
+
+		// Exceeds the max percentage that a slot can be filled
+		asm_data.ingredientInv[0].count = 2;
+		EXPECT_FALSE(CanInsertAssemblyMachine(args));
+
+		// Exceeds max stack size
+		recipe_pack.item1->stackSize    = 50;
+		asm_data.ingredientInv[0].count = 50;
+		EXPECT_FALSE(CanInsertAssemblyMachine(args));
```

Comment:
Does not generate a test failure from exceeding max stack size as it could also be exceeding the max amount a slot can be filled